doc/md/guides/mysql/cd-101.md

```diff
@@ -0,0 +1,156 @@
+---
+id: mysql-cd-101
+title: Detecting drops of foreign-key constraints on MySQL
+slug: /guides/mysql/checks/CD101
+tags: [linting, ci, destructive-change, mysql, drop-foreign-key]
+---
+
+## Introduction
+
+This document describes how Atlas's linter can be used to warn about dropping
+foreign key constraints from a MySQL database.
+
+Let's demonstrate by creating a database with 2 tables - users and pets:
+```sql
+CREATE TABLE users (
+	    id bigint NOT NULL AUTO_INCREMENT,
+	    name varchar(255) NOT NULL,
+	    PRIMARY KEY (id)
+	);
+
+CREATE TABLE pets (
+        id bigint NOT NULL AUTO_INCREMENT,
+        name varchar(255) NOT NULL,
+        owner_id bigint,
+        PRIMARY KEY (id),
+        CONSTRAINT owner_id FOREIGN KEY (owner_id) REFERENCES users (id)
+    );
+```
+
+Since we declared the foreign key constraint on the pets table,
+the database will validate any data inserted to it - blocking insertion
+of rows where the `owener_id` does not correspond to an `id` of an existing user.
+
+Now, let's add a new migration, dropping this constraint as described in the
+[MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html#:~:text=the%20foreign%20key.-,Dropping%20Foreign%20Key%20Constraints,-You%20can%20drop)
+:
+```sql
+ALTER TABLE pets DROP FOREIGN KEY owner_id;
+```
+
+This action, while won't cause data loss, can have some unpredictable side effects
+(as described below). This is why we may want to run some inspection to provide us 
+with some alerts when attempting to drop a foreign-key constraint.
+This can be achieved with Atlas's linter by running:
+
+```bash
+atlas migrate lint --dir file://migrations --dev-url "mysql://root:pass@localhost:3306/dev" --latest 1
+```
+
+In our example, the last command will produce the following output:
+```bash
+20230305125128.sql: constraint deletion detected:
+
+	L1: Dropping foreign-key constraint "owner_id"
+```
+
+([Click here](/versioned/lint) for more information about how to use the atlas cli linter).
+
+## Impact of dropping foreign key constraints
+
+Like mentioned above, dropping a foreign key constraint from a table is not a destructive
+action, meaning it won't cause data loss. And yet, it can have some negative impact since 
+it influences the business logic of our system.
```

Comment:
```suggestion
it can influence the business logic of our system.
```
The BL is not necessarily encoded here as well. Also, add a small example maybe.